docs_src/threat-models/secret-store/vault_master_key_encryption.md

```diff
@@ -0,0 +1,118 @@
+# Vault Master Key Encryption Feature
+
+## Introduction
+
+The EdgeX secret store threat model calls out a particular aspect
+of the Vault-based secret store architecture
+upon which the whole EdgeX secret store depends: the Vault master key.
+Because plaintext storage of the Vault master key at rest would be
+a known [security weakness](https://cwe.mitre.org/data/definitions/313.html),
+the high level design calls for the Vault master key to be encrypted on storage.
+
+One way of doing this would be to simply encrypt the whole drive
+upon which the Vault master key is stored.
+This is a good solution:
+it would encrypt not only the Vault master key,
+but also other part of the system to harden them against
+offline tampering and information disclosure risks.
+This solution also has drawbacks as well:
+whole volume encryption may slow down boot times
+and have a runtime performance impact on constrained
+devices without hardware-accelerated crypto.
+
+The Vault Master Key Encryption feature of EdgeX enables
+a system designer to specifically target encryption of the Vault master key,
+and enables a variety of flexible use cases that are not tied to volume encryption
+such as key escrow (where a key is stored on another machine on the network),
+smart cards or USB HSMs (where a key us stored in a dongle or chip card),
+or TPM (security hardware found on many PC-class motherboards).
+
+## Internal design
+
+As stated in the high level design,
+an [RFC-5869](https://tools.ietf.org/html/rfc5869) key derivation function (KDF)
+is used to produce a set of wrapping keys
+that are used by the vault-worker process to encrypt the Vault master key.
+
+An RFC-5869 KDF requires three inputs.
+A change to any input results in a different output key:
+
+* **Input keying material (IKM).**
+  It need not (but should be) cryptographically strong, and is the "secret" part of the KDF.
```

Comment:
It need not (but should be) -> It need not be (but should be)
or
It need not (but should) be